extension/packages/nextjs/app/eip-712/page.tsx

```diff
@@ -0,0 +1,154 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import type { NextPage } from "next";
+import useSWRMutation from "swr/mutation";
+import { SignTypedDataReturnType } from "viem/accounts";
+import { useAccount, useSignTypedData, useVerifyTypedData } from "wagmi";
+import {
+  EIP_712_DOMAIN,
+  EIP_712_TYPE,
+  VerifyRequestBody,
+  generateMessageToBob,
+} from "~~/utils/eip-712";
+import { notification } from "~~/utils/scaffold-eth";
+import { postMutationFetcher } from "~~/utils/swr";
+
+const Home: NextPage = () => {
+  const { address: connectedAddress } = useAccount();
+  const [signature, setSignature] = useState<SignTypedDataReturnType>();
+  const { signTypedDataAsync } = useSignTypedData();
+
+  const [name, setName] = useState("");
+  const [message, setMessage] = useState("");
+
+  const { trigger: verify } = useSWRMutation(
+    "/api/verify",
+    postMutationFetcher<VerifyRequestBody>
+  );
+
+  const typedData = {
+    domain: EIP_712_DOMAIN,
+    types: EIP_712_TYPE,
+    primaryType: "Mail",
+    message: generateMessageToBob({
+      fromName: name,
+      fromAddress: connectedAddress,
+      message,
+    }),
+  } as const;
+
+  const { data: verifiedOnFrontend } = useVerifyTypedData({
+    ...typedData,
+    address: connectedAddress,
+    signature,
+  });
+
+  const signTypedData = async () => {
+    const signature = await signTypedDataAsync(typedData);
+
+    setSignature(signature);
+    return signature;
+  };
+
+  const verifyOnFrontend = () => {
+    if (verifiedOnFrontend) {
+      notification.success("Success!");
+      return;
+    }
+
+    notification.error("Verification failed");
+  };
+
+  const verifyOnBackend = async () => {
+    if (!connectedAddress) {
+      notification.info("Connect your wallet");
+      return;
+    }
+
+    if (!signature) {
+      notification.info("Typed data is not signed");
+      return;
+    }
+
+    try {
+      await verify({
+        fromName: name,
+        message,
+        signature,
+        signer: connectedAddress,
+      });
+      notification.success("Success!");
+    } catch (err) {
+      console.log(err);
+      notification.error("Verification failed");
```

Comment:
nitpick: 

We could use `getParsedError`  here too if we want, it will parse the message which backend sends

Since in `postMutationFetcher` we are doing `throw new Error(data.error)`

Reply:
Thanks, updated [981741a](https://github.com/scaffold-eth/create-eth-extensions/pull/3/commits/981741a88e885d1ec1be0d103f325a5bbbbe1211)